packages/web-fragments/package.json

```diff
@@ -35,15 +35,19 @@
 	],
 	"scripts": {
 		"dev": "vite",
-		"build": "vite build",
-		"types:check": "tsc --noEmit"
+		"build": "pnpm types && vite build",
+		"types:elements": "tsc -p src/elements",
+		"types:gateway": "tsc -p src/gateway",
+		"types:middlewares": "tsc -p src/gateway/middlewares/tsconfig.cloudflare.json",
```

Comment:
nit: what about putting the pages middleware logic in a directory, have a standard tsconfig and not having to specify one directly here?
e.g.:
```
"types:middlewares": "tsc -p src/gateway/middlewares/cloudflare-pages",
```

I think It'd feel more consistent with the other entries here and would also future proof the structure here (otherwise in the future, would the `middlewares` directory have a mishmash of ts and tsconfig files?)